boards/common/stm32/Kconfig.g0

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) 2020 Inria
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+menu "STM32 G0 clock configuration"
+    depends on CPU_FAM_G0
+
+choice
+bool "Clock source selection"
+default USE_CLOCK_PLL
+
+config USE_CLOCK_PLL
+    bool "PLL"
+
+config USE_CLOCK_HSE
```

Comment:
Here and bellow: any reason not to add the configuration for `CLOCK_HSE`?

Reply:
I added a config for CLOCK_HSE initially but thought there was no reason for it to be customizable via Kconfig. This is something that is specific to the board, so I think there's no reason to expose it in Kconfig since it's not configurable and not useful for other configurations.

Note that compared to the current situation in master, I split the semantic of CLOCK_HSE is 2:
- BOARD_HAS_CLOCK_HSE is a boolean that tells if the board provides an HSE.
- CLOCK_HSE which gives the frequency of external oscillator. In master, when a board doesn't have an HSE, this value is simply set to 0. But in practice this value cannot be 0: for example on stm32gx, HSE must be between 4MHz and 48MHz. So it was not possible to keep that dual semantic with only one variable in Kconfig.